Framework/Algorithms/test/CalcCountRateTest.h

```diff
@@ -0,0 +1,478 @@
+      return;
+
+    for (size_t i = 0; i < testVisWS->getNumberHistograms(); ++i) {
+      const MantidVec &Y = testVisWS->readY(i);
+      double sum = std::accumulate(Y.begin(), Y.end(), 0.);
+      TSM_ASSERT_DELTA("Incorrect counts at index: " +
+                           boost::lexical_cast<std::string>(i),
+                       counts[i], sum, 1.e-6);
+    }
+  }
+
+  void test_visWS_NormalizationFine() {
+
+    auto sws = build_test_ws(true);
+
+    CalcCountRateTester alg;
+    alg.initialize();
+
+    alg.setProperty("NumTimeSteps", 300);
+    alg.setProperty("XResolution", 200);
+
+    alg.setProperty("RangeUnits", "dSpacing");
+
+    alg.setProperty("NormalizeTheRate", true);
+    alg.setProperty("UseLogDerivative", true);
+    alg.setProperty("UseNormLogGranularity", true);
+
+    alg.setProperty("Workspace", sws);
+
+    alg.setOutLogParameters(sws);
+    alg.setSearchRanges(sws);
+    TS_ASSERT_THROWS_NOTHING(alg.setVisWS("testVisWSNorm"));
+
+    auto visNormLog = alg.getVisNormLog();
+    TS_ASSERT_EQUALS(visNormLog.size(), 100);
+    TS_ASSERT_DELTA(visNormLog[10], 2, 1.e-5);
+    auto sum = std::accumulate(visNormLog.begin(), visNormLog.end(), 0.);
+
+    TS_ASSERT_DELTA(sum, 200., 1.e-4);
+  }
+
+  void test_visWS_NormalizationCoarce() {
+
+    auto sws = build_test_ws(true);
+
+    CalcCountRateTester alg;
+    alg.initialize();
+
+    alg.setProperty("NumTimeSteps", 50);
+    alg.setProperty("XResolution", 200);
+
+    alg.setProperty("RangeUnits", "dSpacing");
+
+    alg.setProperty("NormalizeTheRate", true);
+    alg.setProperty("UseLogDerivative", true);
+    alg.setProperty("UseNormLogGranularity", true);
+
+    alg.setProperty("Workspace", sws);
+
+    alg.setOutLogParameters(sws);
+    alg.setSearchRanges(sws);
+    TS_ASSERT_THROWS_NOTHING(alg.setVisWS("testVisWSNorm"));
+
+    auto visNormLog = alg.getVisNormLog();
+    TS_ASSERT_EQUALS(visNormLog.size(), 50);
+    TS_ASSERT_DELTA(visNormLog[10], 4, 1.e-5);
+    auto sum = std::accumulate(visNormLog.begin(), visNormLog.end(), 0.);
+
+    TS_ASSERT_DELTA(sum, 200., 1.e-4);
+  }
+
+  void test_visWS_Normalized() {
+
+    auto sws = build_test_ws(true);
+
+    CalcCountRateTester alg;
+    alg.initialize();
+
+    alg.setProperty("NumTimeSteps", 50);
+    alg.setProperty("XResolution", 200);
+    alg.setProperty("VisualizationWs", "testVisWSNormalized");
+
+    alg.setProperty("RangeUnits", "dSpacing");
+
+    alg.setProperty("NormalizeTheRate", true);
+    alg.setProperty("UseLogDerivative", true);
+    alg.setProperty("UseNormLogGranularity", true);
+
+    alg.setProperty("Workspace", sws);
+
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+
+    API::MatrixWorkspace_sptr testVisWS = alg.getProperty("VisualizationWs");
+    TS_ASSERT(testVisWS);
+    if (!testVisWS)
+      return;
+    TS_ASSERT_EQUALS(testVisWS->getNumberHistograms(), 50);
+    const MantidVec &X = testVisWS->readX(0);
+    const MantidVec &Y = testVisWS->readY(0);
```

Comment:
To my understanding, these new histogram methods are not yet exposed to Python, so it remains to be seen how they will work there.